### DISCUSSION TITLE
How does DaisyUI show/hide functionality work?

### DISCUSSION THREAD

**Discussion by acalvino4** [saadeghi/daisyui#1439] [Q&A] [answered]
For example, on [the dropdown componenet](https://daisyui.com/components/dropdown/#dropdown-menu), clicking the button reveals the menu. But how? When I inspect the menu on that page, it has `visibility: hidden` and `opacity: 0`. And when I click the button, that doesn't change or get overridden but I still see the menu. What is causing the menu to show up in spite of
```css
.dropdown-content {
    visibility: hidden;
    opacity: 0;
}
```

My first though is that there is some custom js that adds `.dropdown-open` to the dropdown element, but nope, no classes are being added, and no inline styles either. How is this working?

**Comment by acalvino4** [comment]
On a broader point, the reason I'm wondering is because there is no clear documentation that I could find delineating what aspects Daisy takes care of, and what I need my own js for. Is there a standard way of toggling things in daisy, such as a class I'm supposed to add/remove?

**Comment by sethcalebweeks** [comment, marked as answer]
It depends on the component. Some of them use an extra checkbox input to toggle the sibling. For example, the modal component is made up of a label with an associated checkbox next to the modal. Note that in order for this to work, the `for` attribute of the label has to match the `id` of the checkbox, and the modal container has to be directly after the checkbox in the markup so that the sibling selector will work.

Other components make use of the focus state, such as the select (dropdown). When the button is clicked, the entire container in which the button and dropdown are in receive focus, and the dropdown is displayed. Clicking outside the container causes it to lose focus and hide the dropdown.

_To be honest, I'm not really sure how these components are meant to be used. They provide a nice reference, but if you want to use them with Vue, React, or any other JavaScript, the CSS hacks kind of get in the way. For example, you would want the dropdown to close after clicking on an option, but the only way to get DaisyUI to close it is to get rid of the focus. You might decide to focus on the button part of the dropdown itself, but that would not work either, so you'll have to find somehting else on the page to focus on._

**Reply by sethcalebweeks** [reply]
Okay, just realized that there is a separate select component that just adds styling to the button part of the select, and rounds the corners of the native dropdown. I confused the dropdown component with the select. If you use the dropdown component for what it is meant for (navigation), then clicking on an item will cause it to lose focus, and it will work as expected.
